src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Sdk.Before.targets

```diff
@@ -19,4 +19,13 @@
     <_MauiUsingDefaultRuntimeIdentifier>true</_MauiUsingDefaultRuntimeIdentifier>
   </PropertyGroup>
 
+  <!--
+    Workaround for https://github.com/NuGet/Home/issues/6461
+    By default, Android and iOS set the Version property, so we need to do the same for Windows.
+    This also has to be done outside of the NuGets as this affects the NuGet restore.
+  -->
+  <PropertyGroup Condition=" '$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == 'windows' and '$(OutputType)' == 'WinExe' ">
+    <Version Condition=" $([System.Version]::TryParse ('$(ApplicationDisplayVersion)', $([System.Version]::Parse('1.0')))) ">$(ApplicationDisplayVersion)</Version>
+  </PropertyGroup>
```

Comment:
This used to be in WinUI targets:

* https://github.com/dotnet/maui/blob/net7.0/src/Workload/Microsoft.Maui.Sdk/Sdk/WinUI.targets

But I guess that won't work, because this would only work after a restore now:

* https://github.com/dotnet/maui/blob/main/src/Core/src/nuget/buildTransitive/WinUI.targets

There are 2 things different with this change than before:
* This is in a `*.Before.targets` and it used to be `*.After.targets`. Is the ordering right?
* The condition is a little different than what imported `WinUI.targets`:

https://github.com/dotnet/maui/blob/8b8293f2b98c3a640643e20a6f28627eff8198a6/src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Core.Sdk.After.targets#L14

If this was moved to some `*.After.targets` would be be able to use the same `Condition` as before? `$(TargetPlatformIdentifier)` is probably blank at the point you have this.

Reply:
I think using `GetTargetPlatformIdentifier` is the most correct way. I believe? Order does not really matter as it all comes after the workload targets - and ios/android seems to be in this spot.

`'$(WindowsAppSDKWinUI)' == 'true'` This is the real issue actually. when you did a PR back in the day, this was not there. I added it at some point and it broke. So now I am moving this out - and to be consistent with android and ios I am using the Before spot - which is a bad name really - it is just immediate. No idea why it is "before"...